### PR TITLE
DT-6175 part 2 itinerary page and its sub component refactoring

### PR DESCRIPTION
**Pull request by vesameskanen**
- Fix all prop type warnings related to wrong prop type definitions, missing props etc.
- Add valid key to all components in lists (fixes debugger warnings)
- Validate prop values before passing them
- Code simplification and cleanup 

After these fixes, debugger gives only relay related warnings from the itinerary page.

### CODE REVIEW

File: app/component/ItineraryDetails.js [hunk 1]
```diff
@@ -256,88 +261,75 @@ class ItineraryDetails extends React.Component {
...
+              isMobile={isMobile}
+              hideBottomDivider={isMobile && shouldShowFarePurchaseInfo(
+                config,
+                breakpoint,
+                fares,
+              )}
+            />,
             showRentalBikeDurationWarning && (
               <VehicleRentalDurationInfo
+		key="rentaldurationinfo"
```
Comment (optionsome) [inline app/component/ItineraryDetails.js:308]: I wonder what case we should actually use for keys, seems like we use like 5 different variations.

File: app/component/ItineraryPage.js [hunk 2]
```diff
@@ -889,35 +798,56 @@ class ItineraryPage extends React.Component {
...
+    } else if (!isEmpty(state.itineraryTopics)) {
+      this.stopClientAndUpdateTopics();
     }
-  }
+    if (
+      hash === 'parkAndRide' &&
+      !state.isFetchingWalkAndBike &&
+      !Array.isArray(state.parkRidePlan?.itineraries)
+    ) {
+      this.toggleStreetMode(''); // go back to showing normal itineraries
```
Comment (optionsome) [inline app/component/ItineraryPage.js]: I know this isn't introduced here but empty string always feels a bit wrong.

File: app/component/ItineraryPage.js [hunk 3]
```diff
@@ -1012,19 +923,24 @@ class ItineraryPage extends React.Component {
...
+                  !Number.isNaN(windSpeed) &&
+                  !Number.isNaN(iconIndex)
+                ) {
+                  weatherData = {
+                    weatherHash,
+                    time,
+                    temperature,
+                    windSpeed,
+                    // Icon spec: www.ilmatieteenlaitos.fi/latauspalvelun-pikaohje -> Sääsymbolien selitykset ennusteissa
+                    iconId: checkDayNight(iconIndex, timem, from.lat, from.lon),
```
Comment (optionsome) [inline app/component/ItineraryPage.js]: timem?

**Review by optionsome — CHANGES_REQUESTED** [review]
Timepicker in front page is broken

File: app/component/ItineraryDetails.js (hunk 1 above)
Comment (vesameskanen) [reply app/component/ItineraryDetails.js:308]: Converted all to lower case

File: app/component/ItineraryPage.js (hunk 2 above)
Comment (vesameskanen) [reply app/component/ItineraryPage.js]: I agree. Cleanup done.

**Review by optionsome — APPROVED** [review]

**Pull request merged**